llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -888,6 +888,14 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
     return;
   }
 
+  if (auto *Steps = dyn_cast<VPScalarIVStepsRecipe>(&R)) {
+    if (Steps->getParent()->getPlan()->isUnrolled() &&
+        Steps->getNumOperands() == 2 && vputils::onlyFirstLaneUsed(Steps)) {
```

Comment:
Isn't `Steps` guaranteed to have two operands? The constructor for VPScalarIVStepsRecipe seems to suggest that.

Reply:
If the VPlan gets unrolled, the unrolled part will be added as extra operand. No operand implies being for the first part and we cannot simplify other parts.

I added a comment to clarify (and also added a hopefully more descriptive way to check for part 0)